parsers/IN_GJ.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+import re
+import string
+
+import requests
+from arrow import get
+from requests import Session
+from parsers import countrycode
+from parsers.lib import web
+from parsers.lib import IN
+from operator import itemgetter
+
+station_map = {
+    "hydro": ["Ukai  (Hydro)", "Kadana (Hydro)", "SSP(RBPH)"],
+    "thermal": ["Ukai(3-5)+Ukai6", "Wanakbori", "Gandhinagar", "Sikka(3-4)", "KLTPS(1-3)+KLTPS4",
+                "SLPP(I+II)", "Akrimota", "JHANOR"],
+    "wind": ["GMR"],
+    "gas": ["Utran(Gas)(II)", "Dhuvaran (Gas)(I)+(II)", "GIPCL(I)", "GIPCL(II)", "GSEG(I+II)", "GPPC", "CLPI",
+            "KAWAS", "Sugen+ Unosgn"],
+    "nuclear": ["KAPP"],
+    "coal": ["TPAECo", "EPGL(I+II)", "Adani(I+II+III)", "BECL(I+II)", "CGPL"]
+}
+
+
+def fetch_data(country_code):
+    countrycode.assert_country_code(country_code, 'IN-GJ')
+
+    solar_html = web.get_response_soup(country_code, 'https://www.sldcguj.com/RealTimeData/GujSolar.php', session)
+    wind_html = web.get_response_soup(country_code, 'https://www.sldcguj.com/RealTimeData/wind.php', session)
+
+    india_date = get(solar_html.find_all('tr')[0].text.split('\t')[-1].strip(), 'D-MM-YYYY h:m:s')
+
+    solar_value = float(eval(solar_html.find_all('tr')[-1].find_all('td')[-1].text.strip()))
+    wind_value = float(eval(wind_html.find_all('tr')[-1].find_all('td')[-1].text.strip()))
+
+    hydro_value = thermal_value = gas_value = nuclear_value = coal_value = 0.0
+
+    value_map = {
+        "date": india_date.datetime,
+        "solar": solar_value,
+        "hydro": hydro_value,
+        "thermal": thermal_value,
+        "wind": wind_value,
+        "gas": gas_value,
+        "nuclear": nuclear_value,
+        "coal": coal_value
+    }
+
+    cookies_params = {
```

Comment:
These are only used in the other_html variable which is commented out. Unless there is a need for them later both can be removed.

Reply:
I prefer to keep it for now. I might add few changes later on.